### PR TITLE
Updated the recipe, and made use of the new matrix extension capabilities of conda-smithy.

### PR DESCRIPTION
**Pull request by pelson**
Probably going to make the linter crash (it can't handle ``os.environ`` yet, PR incoming).

### CODE REVIEW

**Comment by pelson** [comment]
P.S. Linter fixed in https://github.com/conda-forge/conda-smithy/pull/167 and will be available after v0.10.3.

File: conda-forge.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+matrix:
+  - [[VC_VERSION, 9]]
+  - [[VC_VERSION, 10]]
+  - [[VC_VERSION, 14]]
```
Comment (jakirkham) [inline conda-forge.yml]: Where have you been hiding all these sweet features?

**Comment by jakirkham** [comment]
Loving the simplicity of the new recipe. 👍

File: conda-forge.yml (hunk 1 above)
Comment (pelson) [inline conda-forge.yml]: :smile: - https://github.com/conda-forge/conda-smithy/pull/159 :wink:

**Comment by msarahan** [comment]
Nice work, @pelson

**Pull request merged**